File: SEXTO/ACTF1.c
//hacer algo con estructuras, vectores y funciones con datos de los pilotos de f1 
#include <stdio.h>
#include <stdlib.h>
#define LIMITE 50

typedef struct pilotos{
    char nombre[LIMITE];
    char apellido[LIMITE];
    int numAuto;
    char escuderia[LIMITE];
    int puntCampPilotos;
    //int puntCampConstr;
    struct pilotos *next;
}pilotos_t;

typedef enum{
    SALIR,//0
    AGREGARPILOTOS,//1
    MOSTRARPILOTOS,//2
    BUSCARPILOTOS,//3
    MODIFICARPILOTOS,//4
    ELIMINARPILOTOS,//5
}opciones_t;

int inicio();
pilotos_t* agregarPiloto(pilotos_t *lista);
void mostrarPiloto(pilotos_t *lista);
void liberar (pilotos_t *lista);
void buscarPiloto(pilotos_t *lista, int numeroBuscar);
void modificarPuntos(pilotos_t *lista, int numeroMod);

int main()
{
    int opción; 
    pilotos_t *lista = NULL;
    do
    {
        opción=inicio();
        switch (opción) 
        {
            case AGREGARPILOTOS:
                lista = agregarPiloto(lista);
                break;
            case MOSTRARPILOTOS:
                mostrarPiloto(lista);
                break;
            case BUSCARPILOTOS:
                int numeroBuscar;
                printf("Ingrese el número de auto a buscar:\n");
                scanf("%d", &numeroBuscar);
                buscarPiloto(lista, numeroBuscar);
                break;
            case MODIFICARPILOTOS:
                int numeroMod;
                printf("Ingrese el número de auto del piloto a modificar:\n");
                scanf("%d", &numeroMod);
                modificarPuntos(lista, numeroMod);
                break;
            case ELIMINARPILOTOS:
                //funcion;
                break;
        }
    }while (opción!=SALIR);
    liberar(lista);
    return 0;
}

    int inicio()
    {
    int opcion;
    printf("Bienvenido! Este programa es una base de datos de los pilotos de F1\n");
    printf("Ingrese:\n");
    printf("(1):Para agregar un piloto \n");
    printf("(2):Mostrar pilotos\n");
    printf("(3):Buscar un piloto\n");
    printf("(4):Modificar los puntos de los campeonatos\n");
    printf("(5):Eliminar un piloto\n");
    printf("(0):SALIR\n");
    scanf("%d",&opcion);
    return opcion;
    }

pilotos_t * agregarPiloto(pilotos_t * lista)
    {
    pilotos_t *piloto_ptr = (pilotos_t *)malloc(sizeof(pilotos_t));
    if (piloto_ptr == NULL) {
        printf("No hay más memoria\n");
        return lista;
    }else{
        printf("ingrese el nombre del piloto:\n");
        scanf("%s",piloto_ptr->nombre);
        printf("ingrese el apellido del piloto:\n");
        scanf("%s",piloto_ptr->apellido);
        printf("ingrese el número del auto del piloto:\n");
        scanf("%d",&piloto_ptr->numAuto);
        printf("ingrese la escudería del piloto:\n");
        scanf("%s",piloto_ptr->escuderia);
        printf("ingrese los puntos en el campeonato de pilotos:\n");
        scanf("%d",&piloto_ptr->puntCampPilotos);
        //printf("ingrese los puntos en el campeonato de constructores:\n");
        printf("Piloto agregado correctamente!\n\n");
        piloto_ptr->next = lista;
        return piloto_ptr;
        }
    }
    
void mostrarPiloto(pilotos_t *lista)
    {
        while (lista != NULL)
        {
        printf("Nombre: %s\n", lista->nombre);
        printf("Apellido: %s\n",lista->apellido);
        printf("Número de auto: %d\n", lista->numAuto);
        printf("Escudería: %s\n", lista->escuderia);
        printf("Puntos campeonato de pilotos: %d\n\n", lista->puntCampPilotos);
        //printf("Puntos campeonato de constructores: %d\n\n", lista->puntCampConstr);
        lista = lista->next;
        }
    }

void buscarPiloto(pilotos_t *lista, int numeroBuscar)
{
    while (lista != NULL)
    {
        if (lista->numAuto == numeroBuscar)
        {
            printf("Piloto encontrado:\n\n");
            printf("Nombre: %s\n", lista->nombre);
            printf("Apellido: %s\n", lista->apellido);
            printf("Número de auto: %d\n", lista->numAuto);
            printf("Escudería: %s\n", lista->escuderia);
            printf("Puntos campeonato de pilotos: %d\n\n", lista->puntCampPilotos);
        }
        lista = lista->next;
    }
}

void modificarPuntos(pilotos_t *lista, int numeroMod)
{
    while (lista != NULL)
    {
        if (lista->numAuto == numeroMod)
        {
            printf("Piloto encontrado:\n");
            printf("Nombre: %s %s\n", lista->nombre, lista->apellido);
            printf("Puntos actuales: %d\n", lista->puntCampPilotos);
            printf("Ingrese los nuevos puntos del campeonato: ");
            scanf("%d", &lista->puntCampPilotos);
            printf("Puntos actualizados\n\n");
            return;
        }
        lista = lista->next;
    }
}

void liberar (pilotos_t *lista)
{
    pilotos_t *aux;
    while(lista!=NULL)
    {
        aux=lista;
        lista=lista->next;
        free(aux);
    }
}

File: SEXTO PYTHON/actF1.py
pilotos = []

def mostrarMenu():
    print("\nBASE DE DATOS F1")
    print("1. agregar piloto")
    print("2. ver pilotos")
    print("3. buscar piloto")
    print("4. modificar piloto")
    print("5. eliminar piloto")
    print("6. salir")

def agregarPiloto():
    nombre = input("\nnombre: ")
    apellido = input("apellido: ")
    escuderia = input("escudería: ")
    numero = input("número de auto: ")
    puntos = input("puntos en el campeonato: ")

    piloto = {
        "nombre": nombre,
        "apellido": apellido,
        "escuderia": escuderia,
        "numero": numero,
        "puntos": puntos
    }

    pilotos.append(piloto)
    print("piloto agregado\n")

def mostrarPilotos():
    if len(pilotos) == 0:
        print("no hay pilotos cargados\n")
    else:
        print("\nPILOTOS:")
        i = 1
        for piloto in pilotos:
            print("\nPILOTO" + " " + str(i))
            print(piloto["nombre"] + " " + piloto["apellido"])
            print("escudería: " + piloto["escuderia"])
            print("número: " + piloto["numero"])
            print("puntos: " + piloto["puntos"])
            i += 1

def buscarPiloto():
    nombreBuscado = input("ingrese el nombre a buscar: ")
    encontrado = False
    i = 1
    for piloto in pilotos:
        if piloto["nombre"] == nombreBuscado:
            print("\nPILOTO ENCONTRADO:")
            print("PILOTO " + str(i))
            print(piloto["nombre"] + " " + piloto["apellido"])
            print("escudería: " + piloto["escuderia"])
            print("número: " + piloto["numero"])
            print("puntos: " + piloto["puntos"])
            encontrado = True
        i += 1

    if not encontrado:
        print("no se encontró ningún piloto\n")

def modificarPiloto():
    mostrarPilotos()
    if len(pilotos) == 0:
        return

    num = int(input("\ningrese el número del piloto a modificar: "))
    if 1 <= num <= len(pilotos):
        piloto = pilotos[num-1]
        print("\nelija qué quiere modificar:")
        print("1. nombre")
        print("2. apellido")
        print("3. escudería")
        print("4. número del auto")
        print("5. puntos del campeonato")
        opcion = input("opción: ")

        if opcion == "1":
            piloto["nombre"] = input("nuevo nombre: ")
        elif opcion == "2":
            piloto["apellido"] = input("nuevo apellido: ")
        elif opcion == "3":
            piloto["escuderia"] = input("nueva escudería: ")
        elif opcion == "4":
            piloto["numero"] = input("nuevo número de auto: ")
        elif opcion == "5":
            piloto["puntos"] = input("nuevos puntos: ")
        else:
            print("opción inválida")
            return
        print("piloto modificado\n")
    else:
        print("número inválido\n")

def eliminarPiloto():
    mostrarPilotos()
    if len(pilotos) == 0:
        return

    num = int(input("ingrese el número del piloto a eliminar: "))
    if 1 <= num <= len(pilotos):
        del pilotos[num-1]
        print("piloto eliminado\n")
    else:
        print("número inválido\n")

while True:
    mostrarMenu()
    opcion = input("elija una opción: ")
    if opcion == "1":
        agregarPiloto()
    elif opcion == "2":
        mostrarPilotos()
    elif opcion == "3":
        buscarPiloto()
    elif opcion == "4":
        modificarPiloto()
    elif opcion == "5":
        eliminarPiloto()
    elif opcion == "6":
        print("chauu")
        break
    else:
        print("opción inválida\n")